Adding Stripe information to the system status report

wjrosa opened the pull request:
<!--
Did I add a title? A descriptive, yet concise, title.
-->

<!--
Issue: Link to the GitHub issue this PR addresses (if appropriate).
-->

Slack thread: p1736978643799399-slack-C4JPKKQBF
P2: p8wKgj-7sh-p2

## Changes proposed in this Pull Request:

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for the PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

I am adding specific information for our Stripe extension to the Status Report data. This will make it easier for the happiness and the development teams to provide support and better identify merchant needs.

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

## Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Please follow the following guidelines when writing testing instructions:

- Include screenshots if there is no similar flow in the critical flows: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Critical-flows
- Assume instructions will be copied over to the Release Testing Instructions wiki page: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions
- Assume instructions will be followed by external testers.
- Assume tester does not have intimate knowledge of Stripe.
-->

- Checkout to this branch on your test environment (`dev/adding-stripe-info-to-system-status-page`)
- As a merchant, access WC status page (`wp-admin/admin.php?page=wc-status`)
- Click the `Get system report` button:
<img width="548" alt="Screenshot 2025-02-13 at 11 00 00" src="https://github.com/user-attachments/assets/0d27a4f7-0df0-4520-a718-9a4be62e7a16" />

- Confirm you can see the new Stripe information on it:
<img width="522" alt="Screenshot 2025-02-13 at 11 00 38" src="https://github.com/user-attachments/assets/431d08b4-477f-4f0a-b61a-54904ba390f1" />

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

wjrosa committed:
Adding Stripe information to the system status page

wjrosa committed:
Merge branch 'develop' into dev/adding-stripe-info-to-system-status-page

wjrosa committed:
Specific unit test

wjrosa committed:
Fix missing include

wjrosa committed:
More unit tests

wjrosa committed:
Unknown label when email is not present

wjrosa committed:
Changelog and readme entries

wjrosa committed:
Merge branch 'develop' into dev/adding-stripe-info-to-system-status-page

diegocurbelo committed:
Merge branch 'develop' into dev/adding-stripe-info-to-system-status-page

diegocurbelo reviewed (changes requested):
The code looks good, and it tests great.. but one of the new tests is failing, we might need to replace `assertContains` with `assertStringContainsString` or similar:

```
1) WC_Stripe_Status_Test::test_render_status_report_section
TypeError: PHPUnit\Framework\Assert::assertContains(): Argument #2 ($haystack) must be of type iterable, string given, called in /home/runner/work/woocommerce-gateway-stripe/woocommerce-gateway-stripe/tests/phpunit/test-class-wc-stripe-status.php on line 45
```

---

Also, I noticed the other sections have a tooltip (somewhat redundant), but for consistency we might want to add one too:

![Screenshot_2025-02-17_at_17_58_57](https://github.com/user-attachments/assets/e901af69-96cf-47e1-96ba-b6c35a0a776c)

wjrosa committed:
Merge branch 'develop' into dev/adding-stripe-info-to-system-status-page

wjrosa committed:
Fix tests

wjrosa committed:
Adding title tooltip

wjrosa committed:
Improving overall look

wjrosa committed:
Improving overall look

wjrosa commented:
> The code looks good, and it tests great.. but one of the new tests is failing, we might need to replace `assertContains` with `assertStringContainsString` or similar:
> 
> ```
> 1) WC_Stripe_Status_Test::test_render_status_report_section
> TypeError: PHPUnit\Framework\Assert::assertContains(): Argument #2 ($haystack) must be of type iterable, string given, called in /home/runner/work/woocommerce-gateway-stripe/woocommerce-gateway-stripe/tests/phpunit/test-class-wc-stripe-status.php on line 45
> ```
> 
> Also, I noticed the other sections have a tooltip (somewhat redundant), but for consistency we might want to add one too:
> 
> ![Screenshot_2025-02-17_at_17_58_57](https://private-user-images.githubusercontent.com/407542/414009160-e901af69-96cf-47e1-96ba-b6c35a0a776c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4NDA2MDUsIm5iZiI6MTczOTg0MDMwNSwicGF0aCI6Ii80MDc1NDIvNDE0MDA5MTYwLWU5MDFhZjY5LTk2Y2YtNDdlMS05NmJhLWI2YzM1YTBhNzc2Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxOFQwMDU4MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZDBjZWUyZDUwMzc4ODdmMzZjMTgzNjk0YjFlMTNmNzFjZjViMmZjMmIwNGE3NGEzNjVlYWQ2MTQ1MzQyMmI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Ugf3q-jeZ8ru76Cvy56TfXkaSTBJffYQo82ymKsWPjI)

Thanks for the review, Diego! I have fixed the unit test, added the tooltip, and improved the visuals:
<img width="779" alt="Screenshot 2025-02-17 at 22 05 04" src="https://github.com/user-attachments/assets/a459ab00-6e79-4db0-8bea-4901a682ef1e" />

diegocurbelo committed:
Merge branch 'develop' into dev/adding-stripe-info-to-system-status-page

diegocurbelo reviewed (approved):
The new icons and colors look great!!

LGTM 🚢
